resources/authgear/templates/en/web/authflowv2/enter_password.html

```diff
@@ -53,5 +53,94 @@ <h1 class="screen-title">Enter Password</h1>
     By registering, you agree to the <a href="#">Terms of Service</a> and
     <a href="#">Privacy Policy</a>.
   </p>
+
+  <!-- OTP Input -->
+  <div data-controller="otp-input">
```

Comment:
Can we make it a reusable template? The template should:
  - generate all necessary ui elements.
  - The name / value of resend button, submit button, text input should be configurable
  - The error message should be shown depends on a provided context value (.Error)

Reply:
Ideally in the template it should look like

```
<input data-controller="otp-input" type="text" inputmode="numeric" pattern="\d*" maxlength="6">
```

When the controller `connect()`, it transform the element into a OTP component.

I think the OTP component only deals the OTP input. It does not include the resend button, submit button, etc.

Reply:
I think your comment is about the controller at the input, but I mean on top of it, we want a reusable template which generates all neccesary elements at once so we can reuse it in different screen.
Because these things are always used together and never be separated:
- OTP input
- Resend button
- Error message

Reply:
> Can we make it a reusable template? The template should:

I think generating ui elements with connect should avoid those that will cause layout shift 🤔.

Reply:
```
<!-- in OTP page -->
{{ template "__otp_widget.html" (props) }}

<!-- in __otp_widget.html -->
{{ template "__otp_input.html (props) }}
{{ template "__resend_button.html (props) }}
...

<!-- in __otp_input.html -->
In here you can write out all necessary elements that is required for the input.
But only <input> is visible initially. The interactive part is activated by the connected controller 
```

Reply:
By "generate", it is not neccesary to be generated on client side, maybe like:

```html
{{ define "authflowv2/__otp_input.html" }}
    <div data-controller="otp-input">
      <div class="otp-input h-11">
        <input
          data-otp-input-target="input"
          /* ...other things */
        />
        <div data-otp-input-target="textContainer" class="text-container"></div>
      </div>
      <...error>
      <...resend>
    </div>
{{ end }}
```